pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/observables/ObservablesTNCudaOperator.hpp

```diff
@@ -203,49 +257,229 @@ template <class TensorNetT> class ObservableTNCudaOperator {
+                                                         .getStreamID(),
+                                                     *cublascaller_);
+                                }
+                                tensorDataPtrPerTerm_.emplace_back(
+                                    get_obs_device_ptr_(obsKey));
+                            }
+                        } else {
+                            PL_ABORT("Only one wire observables are supported "
+                                     "for cutensornet v24.03");
+                        }
                     }
+                    modes2_.emplace_back(modes_per_term);
+                    numModes2_.emplace_back(num_modes_per_term);
+
+                    // modes pointer initialization
+                    vector1D<const int32_t *> modesPtrPerTerm;
+                    for (std::size_t tensor_idx = 0;
+                         tensor_idx < modes2_.back().size(); tensor_idx++) {
+                        modesPtrPerTerm.emplace_back(
+                            modes2_.back()[tensor_idx].data());
+                    }
+                    modesPtr2_.emplace_back(modesPtrPerTerm);
+                    tensorDataPtr2_.emplace_back(tensorDataPtrPerTerm_);
                 }
-                tensorDataPtrPerTerm_.emplace_back(get_obs_device_ptr_(obsKey));
             }
+        }
 
-            tensorDataPtr_.emplace_back(tensorDataPtrPerTerm_);
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetCreateNetworkOperator(
+            /* const cutensornetHandle_t */ tensor_network.getTNCudaHandle(),
+            /* int32_t */
+            static_cast<int32_t>(tensor_network.getNumQubits()),
+            /* const int64_t stateModeExtents */
+            reinterpret_cast<int64_t *>(const_cast<std::size_t *>(
+                tensor_network.getQubitDims().data())),
+            /* cudaDataType_t */ tensor_network.getCudaDataType(),
+            /* cutensornetNetworkOperator_t */ &obsOperator_));
 
-            appendTNOperator_(coeff, numTensors, numModes_.back().data(),
-                              modesPtr_.back().data(),
-                              tensorDataPtr_.back().data());
+        if (var_cal) {
+            for (std::size_t term_idx = 0; term_idx < numObsTerms2_;
+                 term_idx++) {
+                appendTNOperator_(coeffs2_[term_idx], numTensors2_[term_idx],
+                                  numModes2_[term_idx].data(),
+                                  modesPtr2_[term_idx].data(),
+                                  tensorDataPtr2_[term_idx].data());
+            }
+        } else {
+            for (std::size_t term_idx = 0; term_idx < numObsTerms_;
+                 term_idx++) {
+                appendTNOperator_(coeffs_[term_idx], numTensors_[term_idx],
+                                  numModes_[term_idx].data(),
+                                  modesPtr_[term_idx].data(),
+                                  tensorDataPtr_[term_idx].data());
+            }
```

Comment:
I.e. can't we have a function spit out `coeffs2_`, etc. if `var_cal` and just call 
```suggestion
          for (std::size_t term_idx = 0; term_idx < numObsTerms_;
               term_idx++) {
              appendTNOperator_(coeffs_[term_idx], numTensors_[term_idx],
                                numModes_[term_idx].data(),
                                modesPtr_[term_idx].data(),
                                tensorDataPtr_[term_idx].data());
```

Reply:
Done!